common/api/src/main/java/com/bazaarvoice/emodb/common/api/Names.java

```diff
@@ -28,4 +36,16 @@ public static boolean isLegalTableName(String table) {
                 !(table.charAt(0) == '.' && (".".equals(table) || "..".equals(table))) &&
                 TABLE_NAME_ALLOWED.matchesAllOf(table);
     }
+
+    /**
+     * Role names mostly follow the same conventions as table names except, since they are only used internally, are
+     * slightly more permissive, such as allowing capital letters.  Although "_" is a valid part of a role name it
+     * can't be used as the first character.
+     */
+    public static boolean isLegalRoleName(String role) {
+        return role != null &&
+                role.length() > 0 && role.length() <= 255 &&
+                !role.startsWith("_") &&
+                ROLE_NAME_ALLOWED.matchesAllOf(role);
```

Comment:
In general Emo uses leading underbars to represent reserved items.  I don't mind removing the restriction for role names, but there needs to be protection against "_" as a group name since that is reserved to indicate the absence of a group.